TableDeck.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Card_Game
+{
+    public class TableDeck : Deck
+    {
+        private ICardsDAO FullDeck { get; set; }
```

Comment:
If there is no additional logic behind get and set it can be a private field not property. Properties are usually used when they are public or when something additional happens when get or set are used